swap/src/bitcoin/wallet.rs

```diff
@@ -122,17 +169,37 @@ impl WatchForRawTransaction for Wallet {
 
 #[async_trait]
 impl GetRawTransaction for Wallet {
-    // todo: potentially replace with option
     async fn get_raw_transaction(&self, txid: Txid) -> Result<Transaction> {
-        Ok(self.inner.get_raw_transaction(txid).await?)
+        self.get_tx(txid)
+            .await?
+            .ok_or_else(|| anyhow!("Could not get raw tx with id: {}", txid))
     }
 }
 
 #[async_trait]
 impl GetBlockHeight for Wallet {
     async fn get_block_height(&self) -> BlockHeight {
+        // todo: create this url using the join() api in the Url type
```

Comment:
Just do it?

Reply:
Doable but not that straightforward. join can error and this function is not meant to fail.